DynamoDbEncryption/dafny/DynamoDbEncryption/src/SearchInfo.dfy

```
@@ -253,6 +269,7 @@ module SearchableEncryptionInfo {
         return Failure(AwsCryptographyMaterialProviders(AwsCryptographyMaterialProviders:=getCacheOutput.error));
       }
```

Add this in the MPL HKeyring getActiveBranchKey and getVersionBranchKey to mitigate risk of swallowing errors
```
// If error is not EntryDoesNotExist, return failure
      if (getCacheOutput.Failure? && !getCacheOutput.error.EntryDoesNotExist?) {
        return Failure(AwsCryptographyMaterialProviders(AwsCryptographyMaterialProviders:=getCacheOutput.error));
      }
```

---

Also fix spec duvet for cache entry with limits where it is added

---

This is not an action item for this PR

---

Resolved here https://github.com/aws/aws-cryptographic-material-providers-library/pull/846